test_scripts/API/VehicleData/WindowStatus/common.lua

```diff
@@ -0,0 +1,454 @@
+--! pAppID: application number (1, 2, etc.)
+--! @return: none
+--]]
+function m.subUnScribeVD(pRPC, pAppID)
+  if not pAppID then pAppID = 1 end
+  local cid = m.getMobileSession(pAppID):SendRPC(pRPC, { windowStatus = true })
+    m.getHMIConnection():ExpectRequest("VehicleInfo." .. pRPC, { windowStatus = true })
+    :Do(function(_,data)
+      m.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", { windowStatus = m.subUnsubParams })
+    end)
+  m.getMobileSession(pAppID):ExpectResponse(cid, { success = true, resultCode = "SUCCESS", windowStatus = m.subUnsubParams })
+  m.getMobileSession(pAppID):ExpectNotification("OnHashChange")
+  :Do(function(_, data)
+    m.setHashId(data.payload.hashID, pAppID)
+  end)
+end
+
+--[[ @checkResumption: function that checks resume of subscription for two applications
+--! @parameters:
+--! pFirstApp: true - in case SDL sends VehicleInfo.SubscribeVehicleData_requset to HMI, otherwise - false
+--! pAppId: application number (1, 2, etc.)
+--! @return: none
+--]]
+function m.checkResumption(pFirstApp, pAppId)
+  if not pAppId then pAppId = 1 end
+  local cid = m.getMobileSession(pAppId):SendRPC("SubscribeVehicleData", { windowStatus = true })
+  if pFirstApp then
+    m.getHMIConnection():ExpectRequest("VehicleInfo.SubscribeVehicleData", { windowStatus = true })
+    :Do(function(_,data)
+      m.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", { windowStatus = m.subUnsubParams })
+    end)
+  end
+  m.getMobileSession(pAppId):ExpectResponse(cid, { success = true, resultCode = "SUCCESS", windowStatus = m.subUnsubParams })
+  m.getMobileSession(pAppId):ExpectNotification("OnHashChange")
+  :Do(function(_, data)
+    m.setHashId(data.payload.hashID, pAppId)
+  end)
+end
+
+--[[ @sendOnVehicleData: Processing OnVehicleData RPC
+--! @parameters:
+--! pData: parameters for the notification
+--! pExpTime: number of notifications
+--! pAppID: application number (1, 2, etc.)
+--! @return: none
+--]]
+function m.sendOnVehicleData(pData, pExpTime, pAppID)
+  if not pExpTime then pExpTime = 1 end
+  if not pAppID then pAppID = 1 end
+  m.getHMIConnection():SendNotification("VehicleInfo.OnVehicleData", { windowStatus = pData })
+  m.getMobileSession(pAppID):ExpectNotification("OnVehicleData", { windowStatus = pData })
+  :Times(pExpTime)
+end
+
+--[[ @checkNotificationIgnored:
+--! @parameters:
+--! pParam:
+--! pData: parameters for the notification
+--! pValue:
+--! pTb: table with windowStatus data
+--! @return: none
+--]]
+function m.checkNotificationIgnored(pParam, pData, pValue, pTb)
+  local params = pTb
+  params[1][pData][pParam] = pValue
+
+  m.getHMIConnection():SendNotification("VehicleInfo.OnVehicleData", { windowStatus = params })
+  m.getMobileSession():ExpectNotification("OnVehicleData", { windowStatus = params })
+  :Times(0)
+end
+
+--[[ @ignitionOff: IGNITION_OFF sequence
+--! @parameters: none
+--! @return: none
+--]]
+function m.ignitionOff()
```

Comment:
I would propose to use simplified version of this function like in common module for Smoke tests